OpenRA.Mods.Common/Scripting/Global/ActorGlobal.cs

```diff
@@ -71,7 +71,19 @@ public int BuildTime(string type)
 			if (!Context.World.Map.Rules.Actors.TryGetValue(type, out ai))
 				throw new LuaException("Unknown actor type '{0}'".F(type));
 
-			return ai.GetBuildTime();
+			var bi = ai.TraitInfoOrDefault<BuildableInfo>();
+
+			if (bi != null)
```

Comment:
Minor style nit: the convention we use in cases like this is to check for null and return zero here, and then have the main logic unindented